packages/terra-form-select/src/search/Frame.jsx

```diff
@@ -265,7 +265,7 @@ class Frame extends React.Component {
       this.hasEscPressed = true;
       event.stopPropagation();
       this.closeDropdown();
-    } else if (!this.state.isOpen && keyCode === KeyCode.KEY_ESCAPE && this.hasEscPressed) {
```

Comment:
why `hasEscPressed` is removed

Reply:
hasEscPressed check will be preventing to clear selected item when dropdown Collapsed state and when pressing ESC key

Reply:
I think that is an expected behavior of. not clearing selected value on esc Key press. selected value should only be cleared only when `allowClear` prop is set to true

Reply:
Updated - https://github.com/cerner/terra-core/pull/3985/commits/52f44959c126a4d24127d87d91bd7aba8a4ed121